skills-kit-library/test/skills-kit-2.0.test.js

```diff
@@ -0,0 +1,512 @@
+const { FilesReader, SkillsWriter, SkillsErrorEnum } = require('./../skills-kit-2.0');
+
+const { Readable } = require('stream');
+const jimp = require('jimp');
+const urlPath = require('box-node-sdk/lib/util/url-path');
+
+const fileId = 34426356747;
+const fileName = 'sampleFileName.mp3';
+const fileSize = 4200656;
+const boxRequestId = '50067dc9-b656-4712-a8fa-7b3dd53848cc_1591214556';
+const skillId = 75;
+const readToken = 'readtoken12345';
+const writeToken = 'writetoken12345';
+
+const eventBody = {
+    id: boxRequestId,
+    skill: {
+        id: skillId
+    },
+    source: {
+        id: fileId,
+        name: fileName,
+        size: fileSize
+    },
+    token: {
+        read: {
+            access_token: readToken
+        },
+        write: {
+            access_token: writeToken
+        }
+    }
+};
+
+const fileContext = {
+    requestId: boxRequestId,
+    skillId,
+    fileId,
+    fileWriteToken: writeToken
+};
+
+// Mock file stream
+let mockFileStream;
+
+describe('SkillsErrorEnum', () => {
+    test('skills errors', () => {
+        expect(SkillsErrorEnum.FILE_PROCESSING_ERROR).toEqual('skills_file_processing_error');
+        expect(SkillsErrorEnum.INVALID_FILE_SIZE).toEqual('skills_invalid_file_size_error');
+        expect(SkillsErrorEnum.INVALID_FILE_FORMAT).toEqual('skills_invalid_file_format_error');
+        expect(SkillsErrorEnum.INVALID_EVENT).toEqual('skills_invalid_event_error');
+        expect(SkillsErrorEnum.NO_INFO_FOUND).toEqual('skills_no_info_found');
+        expect(SkillsErrorEnum.INVOCATIONS_ERROR).toEqual('skills_invocations_error');
+        expect(SkillsErrorEnum.EXTERNAL_AUTH_ERROR).toEqual('skills_external_auth_error');
+        expect(SkillsErrorEnum.BILLING_ERROR).toEqual('skills_billing_error');
+        expect(SkillsErrorEnum.UNKNOWN).toEqual('skills_unknown_error');
+    });
+});
+
+describe('FilesReader', () => {
+    const body = JSON.stringify(eventBody);
+    const reader = new FilesReader(body);
+
+    beforeEach(() => {
+        mockFileStream = new Readable();
+        mockFileStream.push('audiofilestream');
+        mockFileStream.push(null);
+    });
+
+    test('validateSize() should return true for valid file size', () => {
+        expect(reader.validateSize(5)).toEqual(true);
+    });
+
+    test('validateSize() should throw exception for invalid file size', () => {
+        console.error = jest.fn();
+        expect(() => {
+            reader.validateSize(4);
+        }).toThrowError(SkillsErrorEnum.INVALID_FILE_SIZE);
+        expect(console.error).toBeCalled();
+    });
+
+    test('validateFormat() should return true for valid file format', () => {
+        const allowedFileFormatsList = 'flac,mp3,wav';
+        expect(reader.validateFormat(allowedFileFormatsList)).toEqual(true);
+    });
+
+    test('validateFormat() should throw exception for invalid file format', () => {
+        const allowedFileFormatsList = 'jpeg,jpg,png,gif,bmp,tif';
```

Comment:
Should this be `tiff`?

```suggestion
        const allowedFileFormatsList = 'jpeg,jpg,png,gif,bmp,tiff';
```